File: scripts/R/animation.R
#!/usr/bin/Rscript

# Set the Content-Type header
cat("Content-Type: text/html\r\n\r\n")

# Generate HTML
cat("<!DOCTYPE html><html><head><title>Animation</title></head><body>")
cat("<h1>Simple Animation</h1>")
cat("<div id='container' style='position: relative; width: 400px; height: 200px; border: 1px solid #000;'>")
cat("<div id='rectangle' style='position: absolute; width: 50px; height: 50px; background-color: blue;'></div>")
cat("</div>")

# JavaScript animation
cat("<script>")
cat("var container = document.getElementById('container');")
cat("var rectangle = document.getElementById('rectangle');")
cat("var x = 0; // Initial x-coordinate for the rectangle")

cat("function animateRectangle() {")
cat("  if (x >= (container.offsetWidth - rectangle.offsetWidth) || x <= 0) {")
cat("    x = 0; // Reset the position when it reaches the container's edge")
cat("  }")
cat("  x += 2; // Move the rectangle to the right")
cat("  rectangle.style.left = x + 'px';")
cat("  requestAnimationFrame(animateRectangle);")
cat("}")

cat("animateRectangle();")
cat("</script>")

cat("</body></html>")

File: scripts/ruby/animation.rb
#!/usr/bin/env ruby

# Set the Content-Type header
puts "Content-Type: text/html\r\n"
puts "\r\n"

# Generate the HTML page with the animation
puts "<!DOCTYPE html><html><head><title>Shape Animation</title></head><body>"
puts "<h1>Shape Animation</h1>"
puts "<div id='container' style='position: relative; width: 400px; height: 200px; border: 1px solid #000;'>"

# Ruby CGI script for a simple shape animation
puts "<style>"
puts "  .shape { width: 50px; height: 50px; position: absolute; border-radius: 5px; animation: move 3s linear infinite; }"
puts "  .red { background-color: red; }"
puts "  .green { background-color: green; }"
puts "  .blue { background-color: blue; }"
puts "@keyframes move {"
puts "  0% { left: 0; top: 0; }"
puts "  25% { left: 100%; top: 0; }"
puts "  50% { left: 100%; top: 100%; }"
puts "  75% { left: 0; top: 100%; }"
puts "  100% { left: 0; top: 0; }"
puts "}"
puts "</style>"

colors = %w(red green blue)

# Generate and animate colored squares
10.times do
  color = colors.sample
  puts "<div class='shape #{color}'></div>"
end

puts "</div>"
puts "</body></html>"
